### PR TITLE
Coverage, clean and create/recreate scenarios added, tests refactored and fixed, workflows updated.

### LINKED ISSUE #63: Makefile options
Add the following options to Makefile:
- clean: clean the local file tree (remove all unnecessary temp files, e.g. egg, build, diet, __pycach__, .DS_Store, .pyc, .pyo, .tox/*, coverage related, logs, ...)
- create a new environment: cleans, creates a new venv, install the package

### PR DESCRIPTION
**Pull request by panasevychol**
Coverage, clean and create/recreate scenarios added. (closes #63 )

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -141,6 +153,52 @@ liccheck:
...
+.PHONY: rmcache
+rmcache:
+	find . | grep -E "\(__pycache__|\.pyc|\.pyo$)" | xargs rm -rf
+
+# Run outside Pipenv
+# To exit Pipenv run "deactivate" or Control+D
+.PHONY: clean
+clean:
+	rm -rf .tox
+	rm -rf arcturus.egg-info
```
Comment (5A11) [inline Makefile:169]: no need for this

File: Makefile [hunk 2]
```diff
@@ -141,6 +153,52 @@ liccheck:
 copyright-check:
 	python scripts/check_copyright.py
 
+####################
+### Clean and init commands
+####################
+
+.PHONY: rmcache
+rmcache:
+	find . | grep -E "\(__pycache__|\.pyc|\.pyo$)" | xargs rm -rf
```
Comment (5A11) [inline Makefile:162]: Could you please mirror what we have in the aea framework?
Missing many of the same elements here.

File: Makefile [hunk 3]
```diff
@@ -141,6 +153,52 @@ liccheck:
...
+	make rmcache
+	pipenv --rm
+
+.PHONY: create
+create:
+	pipenv install
+
+.PHONY: create-dev
+create-dev:
+	pipenv install --dev
```
Comment (5A11) [inline Makefile:186]: No need for these vanilla ones. Let's just have the options below with the clean.

File: Makefile [hunk 4]
```diff
@@ -141,6 +153,52 @@ liccheck:
...
+
+# Run outside Pipenv
+# To exit Pipenv run "deactivate" or Control+D
+.PHONY: recreate
```
Comment (5A11) [inline Makefile:190]: rename `new_env`

File: Makefile [hunk 5]
```diff
@@ -141,6 +153,52 @@ liccheck:
...
+recreate:
+	make clean
+	make init
+
+.PHONY: recreate-dev
```
Comment (5A11) [inline Makefile:195]: rename `new_env_dev`

File: Makefile [hunk 6]
```diff
@@ -141,6 +153,52 @@ liccheck:
...
+.PHONY: create-dev
+create-dev:
+	pipenv install --dev
+
+# Run outside Pipenv
+# To exit Pipenv run "deactivate" or Control+D
+.PHONY: recreate
+recreate:
+	make clean
+	make init
```
Comment (5A11) [inline Makefile:193]: Mirror new_env in the aea-framework.

File: Makefile [hunk 7]
```diff
@@ -141,6 +153,52 @@ liccheck:
...
+
+.PHONY: recreate-dev
+recreate-dev:
+	make clean
+	make init-dev
```
Comment (5A11) [inline Makefile:198]: Similar to the above `new_env` but the `dev` option.

File: .github/workflows/workflow.yml [hunk 8]
```diff
@@ -22,6 +22,7 @@ jobs:
     - uses: actions/setup-python@master
       with:
         python-version: ${{ matrix.python-version }}
+    # - uses: docker-practice/actions-setup-docker@master
```
Comment (5A11) [inline .github/workflows/workflow.yml:25]: remove if not needed.

File: Makefile [hunk 9]
```diff
@@ -141,6 +153,73 @@ liccheck:
...
+	rm -fr dist/
+	rm -fr .eggs/
+	rm -fr pip-wheel-metadata
+	find . -name '*.egg-info' -exec rm -fr {} +
+	find . -name '*.egg' -exec rm -fr {} +
+
+.PHONY: clean-docs
+clean-docs:
+#   Update when docs PR is merged to avoid conflicts
+# 	rm -fr docs/build/
```
Comment (5A11) [inline Makefile:175]: Update when docs PR is merged

File: Makefile [hunk 10]
```diff
@@ -141,6 +153,73 @@ liccheck:
...
+
+# This can be replaced with rmcache
```
Comment (5A11) [inline Makefile:177]: What do you mean?

File: Makefile [hunk 11]
```diff
@@ -141,6 +153,73 @@ liccheck:
...
+	find . -name '*~' -exec rm -f {} +
+	find . -name '__pycache__' -exec rm -fr {} +
+	find . -name '.DS_Store' -exec rm -fr {} +
+
+.PHONY: clean-test
+clean-test:
+	rm -fr .tox/
+	rm -f .coverage
+	find . -name ".coverage*" -not -name ".coveragerc" -exec rm -fr "{}" \;
+	# rm -fr coverage.xml
```
Comment (5A11) [inline Makefile:191]: remove if not needed.

File: Makefile [hunk 12]
```diff
@@ -141,6 +153,73 @@ liccheck:
...
+	rm -fr coverage_report/
+	rm -fr .hypothesis
+	rm -fr .pytest_cache
+	rm -fr .mypy_cache/
+	find . -name 'log.txt' -exec rm -fr {} +
+	find . -name 'log.*.txt' -exec rm -fr {} +
+
+v := $(shell pip -V | grep virtualenvs)
+
+.PHONY: new-env
```
Comment (5A11) [inline Makefile:201]: change to `new_env`

File: Makefile [hunk 13]
```diff
@@ -141,6 +153,73 @@ liccheck:
...
+	if [ -z "$v" ];\
+	then\
+		pipenv --rm;\
+		pipenv install;\
+		echo "Enter virtual environment with all development dependencies now: 'pipenv shell'.";\
+	else\
+		echo "In a virtual environment! Exit first: 'exit'.";\
+	fi
+
+.PHONY: new-env-dev
```
Comment (5A11) [inline Makefile:212]: change to `new_env_dev`

File: Makefile (hunk 10 above)
Comment (panasevychol) [reply Makefile:177]: Sorry. Had another more unified one-line command. I've removed it so will remove this comment too.

File: Makefile (hunk 12 above)
Comment (panasevychol) [reply Makefile:201]: Ok. Just tried to keep similarity with other commands in this file.

File: .github/workflows/workflow.yml [hunk 14]
```diff
@@ -3,6 +3,7 @@ name: Checks and Tests
 on:
   push:
     branches:
+      - develop
```
Comment (5A11) [inline .github/workflows/workflow.yml:6]: should you not remove master from the line below?

File: Makefile [hunk 15]
```diff
@@ -117,7 +117,7 @@ mypy:
 
 .PHONY: pylint
 pylint:
-	pylint $(PYCOSM_SRC_DIR) $(PYCOSM_TESTS_DIR) $(PYCOSM_EXAMPLES_DIR) setup.py --ignore-paths "cosmpy/protos/"
+	pylint $(PYCOSM_SRC_DIR) $(PYCOSM_TESTS_DIR) $(PYCOSM_EXAMPLES_DIR) setup.py
```
Comment (5A11) [inline Makefile:120]: why delete the ignore-path `protos`

Comment (panasevychol) [reply Makefile:120]: Moved to `.pylintrc`.

File: .github/workflows/workflow.yml (hunk 14 above)
Comment (panasevychol) [reply .github/workflows/workflow.yml:6]: I've re-enabled that for master. The reason: we run this set of checks for each PR and push to develop/master. For release PR I've decided not to repeat this checks but still run it and plus additional pre-release checks (for now integration tests on macOS).

**Comment by 5A11** [comment]
@panasevychol 
For coverage, we need to add cover for the following:
`cosmpy/tx/rest_client.py` lines `62-65` and `100-101`

Then please add everything under `cosmos/protos` to `.coveragerc` so they get ignored.

File: .github/workflows/workflow.yml [hunk 16]
```diff
@@ -3,6 +3,7 @@ name: Checks and Tests
...
       - master
   pull_request:
```
Comment (5A11) [inline .github/workflows/workflow.yml:8]: Why is there nothing under pull request?

**Review by 5A11 — APPROVED** [review]
LGTM

**Pull request merged**